### ISSUE TITLE
rename clusterSex function to general use function

### ISSUE THREAD

**Issue by LiNk-NY** [waldronlab/MultiAssayExperiment#176]
`clusterSex` can really apply to any sample characteristic, not just sex

**Comment by lwaldron** [comment]
Agreed, here is a suggestion for the API:

```
#' Title pDataExperimentConcordance
#'
#' @param object
#' MultiAssayExperiment object
#' @param primary 
#' column name in the pData (optional). If specified, kmeans clusters will be re-named for maximum concordance with this column.
#' @param FUN (default something with kmeans)
#' a function that takes a numeric matrix as input, and returns a numeric vector of class labels as output.
#' FUNargs (default k=3)
#' Arguments passed on to FUN
#' @return
#' a data.frame containing four columns: 1) the assay name (kmeans will be performed for each assay in the MultiAssayExperiment), 2) sample IDs (rownames from pData), 3) the values from pData, and 4) the predictions from kmeans clustering.
```

**Comment by ttriche** [comment]
maybe "clusterOn(data, characteristicRegion, ...)" would be a little less
painful to type :-)


--t

**Comment by lwaldron** [comment]
Heh, yes that would work for me and is descriptive enough.

**Comment by LiNk-NY** [comment]
Issue resolved in f30bb3bb16912db83ba1b93081254d9e4ff1ce2e
